Add documentation for Pulsar OAuth configuration. (#263)
pinot-contrib/pinot-docs: basics/data-import/pinot-stream-ingestion/apache-pulsar.md

### Authentication

> The Pinot-Pulsar connector supports authentication using the security tokens. You can generate the token by following the [official Pulsar documentation](https://pulsar.apache.org/docs/en/security-token-client/). Once generated, you can add the following property to `streamConfigs` to add auth token for each request

Reviewer: ```suggestion
The Pinot-Pulsar connector supports authentication using security tokens. To generate a token, follow the instructions in [Pulsar documentation](https://pulsar.apache.org/docs). Once generated, add the following property to `streamConfigs` to add an authentication token for each request:
```

Reviewer: appears the link is no longer working, added link to top-level docs site

Author: Sure. I was just correcting a typo there. I have found the updated link to the JWT token auth page.

```
"stream.pulsar.authenticationToken":"your-auth-token"
```

### OAuth2 Authentication

> The Pinot-Pulsar connector support authentication using OAuth2, for example if connecting to a StreamNative Pulsar cluster. This is covered in the [official Pulsar documentation](https://pulsar.apache.org/docs/en/security-oauth2/#configure-oauth2-authentication-in-pulsar-clients). Once configured, you can add the following properties to `streamConfigs`:

Reviewer: ```suggestion
The Pinot-Pulsar connector supports authentication using OAuth2, for example, if connecting to a StreamNative Pulsar cluster. For more information, see how to [Configure OAuth2 authentication in Pulsar clients](https://pulsar.apache.org/docs/en/security-oauth2/#configure-oauth2-authentication-in-pulsar-clients). Once configured, you can add the following properties to `streamConfigs`:
```
